osu.Desktop/Updater/AppImageUpdateManager.cs

```diff
@@ -0,0 +1,564 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+#nullable disable
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.Versioning;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using osu.Framework;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Colour;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Graphics.Sprites;
+using osu.Game;
+using osu.Game.Graphics;
+using osu.Game.Online.API;
+using osu.Game.Overlays;
+using osu.Game.Overlays.Notifications;
+using osu.Game.Updater;
+using osuTK;
+using osuTK.Graphics;
+
+namespace osu.Desktop.Updater
+{
+    /// <summary>
+    /// An update manager that shows notifications if a newer release is detected.<para />
+    /// Updates the AppImage and backups the previous version with a .zs-old suffix.
+    /// </summary>
+    [SupportedOSPlatform("linux")]
+    public class AppImageUpdateManager : UpdateManager
+    {
+        /// <summary>
+        /// Implements appimageupdatetool functionality via cli
+        /// </summary>
+        private class AppImageUpdateTool
+        {
+            /// <summary>
+            /// Arguments passed to the appimageupdatetool
+            /// <list type="bullet">
+            /// <item><see cref="CHECK"/><description></description></item>
+            /// <item><see cref="UPDATE"/></item>
+            /// <item><see cref="OVERWRITE"/></item>
+            /// <item><see cref="TOOL_VERSION"/></item>
+            /// <item><see cref="TOOL_HELP"/></item>
+            /// </list>
+            /// </summary>
+            public enum ARGS
```

Comment:
Keep enum names explanatory and in pascal casing, something like `ToolArguments`. Same applies to all other enums and enum members as well.